sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Latest.java

```diff
@@ -0,0 +1,213 @@
+ *  {@literal @}ProcessElement
+ *  public void processElement(ProcessContext c) {
+ *    double val = // ..
+ *    latestValue.addValue(TimestampedValue.of(val, c.timestamp()));
+ *    // ..
+ *  }
+ * }
+ * </code></pre>
+ *
+ * <p>For elements with the same timestamp, the element chosen for output is arbitrary.
+ */
+public class Latest {
+  // Do not instantiate
+  private Latest() {}
+
+  /**
+   * A {@link Combine.CombineFn} that computes the latest element from a set of inputs. This is
+   * particularly useful as an {@link Aggregator}.
+   *
+   * @param <T> Type of input element.
+   * @see Latest
+   */
+  public static class LatestFn<T>
+      extends Combine.CombineFn<TimestampedValue<T>, TimestampedValue<T>, T> {
+    /** Construct a new {@link LatestFn} instance. */
+    public LatestFn() {}
+
+    @Override
+    public TimestampedValue<T> createAccumulator() {
+      return TimestampedValue.of(null);
+    }
+
+    @Override
+    public TimestampedValue<T> addInput(TimestampedValue<T> accumulator,
+        TimestampedValue<T> input) {
+      checkNotNull(accumulator, "accumulator must be non-null");
+      checkNotNull(input, "input must be non-null");
+
+      if (input.getTimestamp().isBefore(accumulator.getTimestamp())) {
+        return accumulator;
+      } else {
+        return input;
+      }
+    }
+
+    @Override
+    public Coder<TimestampedValue<T>> getAccumulatorCoder(CoderRegistry registry,
+        Coder<TimestampedValue<T>> inputCoder) throws CannotProvideCoderException {
+      return NullableCoder.of(inputCoder);
+    }
+
+    @Override
+    public Coder<T> getDefaultOutputCoder(CoderRegistry registry,
+        Coder<TimestampedValue<T>> inputCoder) throws CannotProvideCoderException {
+      checkState(inputCoder instanceof TimestampedValue.TimestampedValueCoder,
+          "inputCoder must be a TimestampedValueCoder");
+
+      TimestampedValue.TimestampedValueCoder<T> inputTVCoder =
+          (TimestampedValue.TimestampedValueCoder<T>) inputCoder;
+      return NullableCoder.of(inputTVCoder.<T>getValueCoder());
+    }
+
+    @Override
+    public TimestampedValue<T> mergeAccumulators(Iterable<TimestampedValue<T>> accumulators) {
+      checkNotNull(accumulators, "accumulators must be non-null");
+
+      TimestampedValue<T> merged = createAccumulator();
+      for (TimestampedValue<T> accum : accumulators) {
+        merged = addInput(merged, accum);
+      }
+
+      return merged;
+    }
+
+    @Override
+    public T extractOutput(TimestampedValue<T> accumulator) {
+      return accumulator.getValue();
+    }
+  }
+
+  /**
+   * Returns a {@link PTransform} that takes as input a {@link PCollection<T>} and returns a
+   * {@link PCollection<T>} whose contents is the latest element according to its event time, or
+   * {@literal null} if there are no elements.
+   *
+   * @param <T> The type of the elements being combined.
+   */
+  public static <T> PTransform<PCollection<T>, PCollection<T>> globally() {
```

Comment:
I think most of my concern here is around the addition of `TimestampFn` as a concept. I think I'd be OK with the signature `PTransform<V, V>` and either (1) removing the ability to use a different `TimestampFn` (since all elements are already timestamped) or (2) using a `SerializableFn`. Basically, I'd like to avoid new kinds of UDFs since those require (potentially) special handling in the graph, etc.